### PR TITLE
fix(www): nextjs usage config markdown

### PR DESCRIPTION
**Pull request by charkour**
## 🎯 Changes

This removes incorrect documentation for fields that no long exist on the `createTRPCNext` `config`-callback. It also fixes a markdown bug with the bullet points. 

Thanks for your hard work! Please make any changes as you see fit. 

Related to: 
- #2734
- #2709
- #2723

<details>
<summary>Photo details of change</summary>

Screenshot before change: 
image

Screenshot after change: 
image

</details>

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/nextjs/introduction.md [hunk 1]
```diff
@@ -212,17 +212,11 @@ export default function IndexPage() {
...
-
-  - `url` your API URL.
+- **Required**:
   - `links` to customize the flow of data between tRPC Client and the tRPC-server. [Read more](../client/links.md).
-
 - Optional:
   - `queryClientConfig`: a configuration object for the React Query `QueryClient` used internally by the tRPC React hooks: [QueryClient docs](https://tanstack.com/query/v4/docs/reference/QueryClient)
-  - `headers`: an object or a function that returns an object of outgoing tRPC requests
   - `transformer`: a transformer applied to outgoing payloads. Read more about [Data Transformers](data-transformers)
-  - `fetch`: customize the implementation of `fetch` used by tRPC internally
```
Comment (charkour) [inline www/docs/nextjs/introduction.md:224]: `headers`, `fetch`, and `AbortController` are managed within `links` now. Is there an update to the documentation you would like to see for that?

**Review by juliusmarminge — APPROVED** [review]

File: www/docs/nextjs/introduction.md (hunk 1 above)
Comment (KATT) [reply www/docs/nextjs/introduction.md:224]: We should do an overhaul of https://trpc.io/docs/v10/links

**Comment by charkour** [comment]
@juliusmarminge,

Is it alright if I merge this? Thanks for the review. I noticed @KATT's suggestion to overhaul [trpc.io/docs/v10/links](https://trpc.io/docs/v10/links), but I think that is best suited for someone with a better understanding of the changes. 

Thanks!

**Review by juliusmarminge — COMMENTED** [review]
Might look a bit better having these as H4's. The nested LI looks a bit odd on mobile at least image

**Comment by juliusmarminge** [comment]
> @juliusmarminge,
> 
> 
> 
> Is it alright if I merge this? Thanks for the review. I noticed @KATT's suggestion to overhaul [trpc.io/docs/v10/links](https://trpc.io/docs/v10/links), but I think that is best suited for someone with a better understanding of the changes. 
> 
> 
> 
> Thanks!

Only @KATT can merge this since all checks haven't run

**Comment by charkour** [comment]
Thanks for the feedback and clarification about merging!

**Review by KATT — APPROVED** [review]

**Pull request merged**